src/ReadLoader.cpp

```diff
@@ -66,6 +66,12 @@ ReadLoader::ReadLoader(
         "Supported file extensions are .fasta, .fa, .FASTA, .FA.");
 }
 
+
+// This is required because unique_ptr has a type completeness requirement
+// during destruction.
+ReadLoader::~ReadLoader() = default;
```

Comment:
https://stackoverflow.com/questions/6012157/is-stdunique-ptrt-required-to-know-the-full-definition-of-t/6089065#6089065 has more information on how shared_ptr and unique_ptr differ for type completeness requirements.